source/StandardFirmata.cpp

```diff
@@ -390,7 +390,7 @@ void sysexCallback(byte command, byte argc, byte *argv)
         case BALENA_SLEEP:
           if (argc > 5) {
             power_struct.sleep_delay = argv[1] * (uint32_t) DELAY_MULTIPLIER; // in seconds
-            power_struct.sleep_period = (argv[5] << 24 | argv[4] << 16 | argv[3] << 8 | argv[2]) * (uint32_t) DELAY_MULTIPLIER; // in seconds
+            power_struct.sleep_period = (argv[5] << 21 | argv[4] << 14 | argv[3] << 7 | argv[2]) * (uint32_t) DELAY_MULTIPLIER; // in seconds
```

Comment:
@rohfle This looks good; thanks for catching this! Can you add another index to argv so that we can ensure we're not eating into the max range for the sleep period? `power_struct.sleep_period` takes a uint32_t so right now we're 3 bits short of max range. Additionally, it's fine that we'll have a largely unused argv[6] as I'll likely use this to later implement the EMU controls for the BGM111.

CC @curcuz

Reply:
Started doing this, then spotted an overflow. DELAY_MULTIPLIER is set to 1000 and the sleep_period only is 32 bits unsigned so that only gives you 22 bits usable space or 48 days sleep time max.

Even if you change sleep_period to 64 bits unsigned, I think the timer compare register is only 32 bits?

Updating code to suit findings

Reply:
You're right it is compared against a 32 bit timer compare register and unfortunately that's the largest timer register we have to work with. We could look into adding additional control by using the pre-scaler for the counter (and pass this via the sysex command) but this introduces more complexity for the user to control timing. Alternatively we could set a larger default pre-scaler value so we can sleep for longer but with less precision.

@curcuz what are your thoughts?

Reply:
@Bucknalla @rohfle I believe 97 days is a reasonable max value

Reply:
@rohfle @Bucknalla  our CI allows deployments only on PRs against the master branch :(

@Bucknalla can you please PR and merge 0.0.2 to master, so that @rohfle can reissue his PR against that and we keep a consistent workflow and we keep his commits in our changelog?